samples/boards/stm32/power_mgmt/stop3/src/main.c

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright (c) 2021 Linaro Limited
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/device.h>
+#include <zephyr/devicetree.h>
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/sys/printk.h>
+#include <stm32u5xx_ll_pwr.h>
+
+#define SLEEP_TIME_MS   2000
+
+static const struct gpio_dt_spec led =
+	GPIO_DT_SPEC_GET(DT_ALIAS(led0), gpios);
+
+int main(void)
+{
+	bool led_is_on = true;
+
+	__ASSERT_NO_MSG(gpio_is_ready_dt(&led));
+
+	printk("Device ready\n");
+	gpio_pin_configure_dt(&led, GPIO_OUTPUT_ACTIVE);
+
+	while (true) {
+		gpio_pin_set(led.port, led.pin, (int)led_is_on);
+		/* In STOP3, GPIOs are disabled. Only pull-up/pull-down can be enabled.
+		 * So we enable pull-up/pull-down based on LED status
+		 */
+		if (led_is_on) {
+			PWR->PDCRB &= ~(uint32_t)(1 << 7);
+			PWR->PUCRB |= (1 << 7);
+		} else {
+			PWR->PUCRB &= ~(uint32_t)(1 << 7);
+			PWR->PDCRB |= (1 << 7);
+		}
+		k_msleep(SLEEP_TIME_MS);
```

Comment:
```suggestion
                /* activate PWR pull-ups/pull-downs */
                LL_PWR_EnablePUPDConfig();
                
		k_msleep(SLEEP_TIME_MS);
```

Reply:
Good point. Initially I called the ``LL_PWR_EnablePUPDConfig()`` in power.c. But I think this should be up to user if they want to use this.

Reply:
It can also be called once at the beginning of the example, since it takes effect only in STOP3 mode

Reply:
Are you sure PWR pull-ups/downs have an effect in STOP3 mode? they only mention standby & shutdown here
https://github.com/STMicroelectronics/stm32u5xx_hal_driver/blob/5313c6919f88e8ec4212da91bbe1146d4262d44c/Inc/stm32u5xx_ll_pwr.h#L2650

Reply:
Also shouldn't `LL_PWR_DisablePUPDConfig()` be called before reconfiguring PUPDs? (I'm just thinking out loud)